src/StoreKeeper/WooCommerce/B2C/Core.php

```diff
@@ -534,26 +527,34 @@ public function ts_apply_min_amount_to_all_shipping_methods() {
      * @param $package
      * @return mixed
      */
-    public function ts_modify_shipping_rates($rates, $package) {
+    public function tsModifyShippingRates($rates, $package): mixed {
+        // Get the formatted cart total using WooCommerce's functions and settings
+        $total = WC()->cart->get_cart_contents_total();
+        $formatted_cart_total = floatval($total); // WooCommerce already formats the price, so we just cast it to float
+
+        // Get the available shipping methods
         $shipping_methods = WC()->shipping->get_shipping_methods();
-        $total = floatval(WC()->cart->subtotal);
-        $formatted_cart_total = number_format($total, 2, ',', '');
 
+        // Prepare the methods data, ensuring 'min_amount' is properly retrieved and formatted
         $methods_data = array_map(function($method) {
-            $min_amount = isset($method->instance_settings['min_amount']) ? $method->instance_settings['min_amount'] : 'N/A';
+            // Retrieve the 'min_amount' setting and ensure it uses float
+            $min_amount = isset($method->instance_settings['min_amount']) ? floatval($method->instance_settings['min_amount']) : __('N/A', 'no-min-amount');
             return [
                 'name' => $method->title,
                 'min_amount' => $min_amount
             ];
         }, $shipping_methods);
 
+        // Loop through the shipping rates to apply free shipping where necessary
         foreach ($rates as $rate_key => $rate) {
             if (isset($rate->method_id, $rate->instance_id)) {
                 foreach ($methods_data as $method_data) {
-                    if ($method_data['name'] === $rate->label && floatval($formatted_cart_total) >= floatval($method_data['min_amount'])) {
-                        $rates[$rate_key]->cost = 0;
-                        $rates[$rate_key]->label .= ': Free Shipping';
-                        break;
+                    // Compare method label and check if cart total is greater than or equal to min_amount
+                    if ($method_data['name'] === $rate->label && $formatted_cart_total >= $method_data['min_amount']) {
+                        $rates[$rate_key]->cost = 0; // Set cost to 0 for free shipping
+                        // Append 'Free Shipping' label with translation
+                        $rates[$rate_key]->label .= __(': Free Shipping', 'free-shipping');
```

Comment:
```suggestion
                        $rates[$rate_key]->label .= ': '.__('Free Shipping', 'free-shipping');
```

Where is `free-shipping` domain defined?